src/power_grid_model/core/buffer_handling.py

```diff
@@ -115,6 +118,8 @@ def _get_raw_attribute_data_view(data: np.ndarray, schema: ComponentMetaData, at
     Returns:
         a raw view on the data set.
     """
+    if schema.dtype[attribute].shape == (3,) and data.shape[-1] != 3:
```

Comment:
May you elaborate on this addition please? Seems confusing to me.

Reply:
if its an asym attribute, its shape would be (3,) in the schema. And the condition is to check if it is not equal to the shape of data.